1. Вероятность того, что стрелок попадет в мишень, выстрелив один раз, равна 0.8. Стрелок выстрелил 100 раз. Найдите вероятность того, что стрелок попадет в цель ровно 85 раз.

In [1]:
from math import factorial, exp
def bernulli(n, k, p):
    comb=factorial(n)/(factorial(k)*factorial(n-k))
    return comb*(p**k)*(1-p)**(n-k)

In [2]:
print(f'P = {bernulli(100,85,0.8): .4f}')

P =  0.0481


Ответ: 4.81%

2. Вероятность того, что лампочка перегорит в течение первого дня эксплуатации, равна 0.0004. В жилом комплексе после ремонта в один день включили 5000 новых лампочек.

a) Какова вероятность, что ни одна из них не перегорит в первый день?

In [3]:
def poisson_distr(m,p,n):
    lamb=p*n
    return exp(-lamb)*(lamb**m)/factorial(m)

In [4]:
print(f'P = {poisson_distr(0,0.0004,5000): .4f}')

P =  0.1353


Ответ: 0.1353%

б) Какова вероятность, что перегорят ровно две?

In [5]:
print(f'P = {poisson_distr(2,0.0004,5000): .4f}')

P =  0.2707


Ответ: 0.2707%

3. Монету подбросили 144 раза. Какова вероятность, что орел выпадет ровно 70 раз?

In [ ]:
print(f'P = {bernulli(144,70,0.5): .4f}')

Ответ: 0.0628%

4. В первом ящике находится 10 мячей, из которых 7 - белые. Во втором ящике - 11 мячей, из которых 9 белых. Из каждого ящика вытаскивают случайным образом по два мяча. Какова вероятность того, что:

In [6]:
def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

In [7]:
n1=combinations(10,2)
n2=combinations(11,2)
print(f'Общее число элементарных исходов для первой корзины: n1 = {n1}\n'
f'Общее число элементарных исходов для второй корзины: n2 = {n2}')

Общее число элементарных исходов для первой корзины: n1 = 45
Общее число элементарных исходов для второй корзины: n2 = 55


a) все мячи белые?;

In [8]:
m1=combinations(7,2)
m2=combinations(9,2)
print(f'Число благоприятных элементарных исходов для события А: m1 = {combinations(7,2)}\n'
      f'Число благоприятных элементарных исходов для события В: m2 = {combinations(9,2)}'
     )

Число благоприятных элементарных исходов для события А: m1 = 21
Число благоприятных элементарных исходов для события В: m2 = 36


In [9]:
print(f'Вероятность наступления события А, Р(А) = {m1/n1: .4f}\n'
      f'Вероятность наступления события В, Р(В) = {m2/n2: .4f}\n'
      f'Вероятность одновременного наступления событий А и В, Р(А+В) = {(m1/n1)*(m2/n2): .4f}'
     )

Вероятность наступления события А, Р(А) =  0.4667
Вероятность наступления события В, Р(В) =  0.6545
Вероятность одновременного наступления событий А и В, Р(А+В) =  0.3055


In [10]:
P=(7/10)*(6/9)*(9/11)*(8/10)
print(f'Вероятность одновременного наступления событий А и В, Р(А+В) = {P: .4f}')

Вероятность одновременного наступления событий А и В, Р(А+В) =  0.3055


Ответ: 0.3055

б) ровно два мяча белые?;

In [11]:
m_A=combinations(7,2)
m_B=combinations(2,2)
m_C=combinations(7,1)*combinations(3,1)
m_D=combinations(9,1)*combinations(2,1)
m_E=combinations(3,2)
m_F=combinations(9,2)
print(f'Число благоприятных элементарных исходов для события А: m_A = {m_A}\n'
      f'Число благоприятных элементарных исходов для события B: m_B = {m_B}\n'
      f'Число благоприятных элементарных исходов для события C: m_C = {m_C}\n'
      f'Число благоприятных элементарных исходов для события D: m_D = {m_D}\n'
      f'Число благоприятных элементарных исходов для события E: m_E = {m_E}\n'
      f'Число благоприятных элементарных исходов для события F: m_F = {m_F}'  
     )

Число благоприятных элементарных исходов для события А: m_A = 21
Число благоприятных элементарных исходов для события B: m_B = 1
Число благоприятных элементарных исходов для события C: m_C = 21
Число благоприятных элементарных исходов для события D: m_D = 18
Число благоприятных элементарных исходов для события E: m_E = 3
Число благоприятных элементарных исходов для события F: m_F = 36


In [12]:
P=(m_A/n1)*(m_B/n2)+(m_C/n1)*(m_D/n2)+(m_E/n1)*(m_F/n2)
print(f'Вероятность того, что будут вытащены ровно 2 белых мяча Р = {P: .4f}')

Вероятность того, что будут вытащены ровно 2 белых мяча Р =  0.2048


Ответ: 0.2048%

в) хотя бы один мяч белый?.

In [13]:
m1=combinations(3,2)
m2=combinations(2,2)
print(f'Число благоприятных элементарных исходов для первой корзины: m1 = {combinations(3,2)}\n'
      f'Число благоприятных элементарных исходов для второй корзины: m2 = {combinations(2,2)}'
     )

Число благоприятных элементарных исходов для первой корзины: m1 = 3
Число благоприятных элементарных исходов для второй корзины: m2 = 1


In [14]:
print(f'Вероятность того, что не будет извлечено ни одного белого мяча Р(А_обр) = {(m1/n1)*(m2/n2): .4f}\n'
      f'Вероятность того, что будет извлечен хотя бы один белый мяч Р(А) = 1 - Р(А_обр) = {1-(m1/n1)*(m2/n2): .4f}'
     )

Вероятность того, что не будет извлечено ни одного белого мяча Р(А_обр) =  0.0012
Вероятность того, что будет извлечен хотя бы один белый мяч Р(А) = 1 - Р(А_обр) =  0.9988


Ответ: 99.88%